web/src/views/FileBrowserView/FileBrowser.vue

```diff
@@ -377,18 +377,38 @@ const isOwner = computed(() => !!(
 ));
 const itemsNotFound = computed(() => items.value && !items.value.length);
 
+function replaceStringsInEndpoint(
+  endpoint: string,
+  substitutions: {[key: string]: string} | undefined,
+) {
+  if (!substitutions) return endpoint;
+  let result = endpoint;
+  Object.entries(substitutions).forEach(([key, value]) => {
+    result = result.replace(key, value);
+  });
+  return result;
+}
+
 function getExternalServices(path: AssetPath) {
   const servicePredicate = (service: Service, _path: AssetPath) => (
     new RegExp(service.regex).test(path.path)
           && _path.asset !== null
           && _path.aggregate_size <= service.maxsize
   );
 
+  const baseApiUrl = process.env.VUE_APP_DANDI_API_ROOT;
+
+  const substitutions = path.asset ? {
+    $asset_s3_download_url$: trimEnd(path.asset.url, '/'),
+    $asset_api_download_url$: `${baseApiUrl}assets/${path.asset.asset_id}/download/`,
```

Comment:
minor suggestion:

```suggestion
  const baseApiUrl = trimEnd(process.env.VUE_APP_DANDI_API_ROOT, '/');

  const substitutions = path.asset ? {
    $asset_s3_download_url$: trimEnd(path.asset.url, '/'),
    $asset_api_download_url$: `${baseApiUrl}/assets/${path.asset.asset_id}/download/`,
```
to avoid assumption of having trailing `/`?